Once-a-while grammar and style review for the Datasource guide (#46128)
quarkusio/quarkus: docs/src/main/asciidoc/datasource.adoc

===== Oracle considerations

As documented in link:https://github.com/quarkusio/quarkus/issues/36265[issue #36265],
Oracle unexpectedly commits uncommitted transactions when closing a connection.
This means that when stopping Quarkus, in-progress transactions might be committed even if they are incomplete.

> Since this behavior is unexpected and could lead to data loss, an interceptor rolls back any unfinished transactions when closing a connection, unless XA transactions are used, in which case the transaction manager handles the rollback.

Reviewer: ```suggestion
Because this behavior is unexpected and can lead to data loss, an interceptor rolls back any unfinished transactions when closing a connection. 
However, if you use XA transactions, the transaction manager handles the rollback.
```

If the behavior introduced in 3.18 causes issues for your workload, disable it by setting the `-Dquarkus-oracle-no-automatic-rollback-on-connection-close` system property to `true`.

> If possible, report your use case in our link:https://github.com/quarkusio/quarkus/issues[issue tracker] so we can adjust this behavior if needed.

Reviewer 2: Two "ifs" in the same sentence?

At any rate I'd remove "if possible". I don't see why it wouldn't possible for some to report that they need this. And if they don't we'll eventually remove this flag, so they really should do it.

For example:

```suggestion
Make sure to report your use case in our link:https://github.com/quarkusio/quarkus/issues[issue tracker] so we can adjust this behavior if needed, for example with more permanent settings.
```

}
----

Alternatively, you may define a xref:cdi.adoc#ok-you-said-that-there-are-several-kinds-of-beans[CDI bean producer] for the default datasource redirecting to the currently active named datasource, so that it can be injected directly, like this:

> 

Reviewer 2: Was this removed for a reason?

Author: Good point. I have it fixed but forgot to apply it. Returning revamped. THX!
